worlds/pokemon_rb/rom.py

```diff
@@ -558,9 +608,16 @@ def generate_output(self, output_directory: str):
     slot_name.replace(">", " ")
     write_bytes(data, encode_text(slot_name, 16, True, True), rom_addresses['Title_Slot_Name'])
 
-    write_bytes(data, self.trainer_name, rom_addresses['Player_Name'])
-    write_bytes(data, self.rival_name, rom_addresses['Rival_Name'])
+    if self.trainer_name == -1:
```

Comment:
like earlier, hopefully the option compare that doesn't rely on random ints to remember works here

Reply:
changed